### ISSUE TITLE
Low priority:  cp -p the /public directory

### ISSUE THREAD

**Issue by scubachristopher** [thumbsup/thumbsup#177]
## Feature request

Since the content of this directory and subs are not generated, cp -p would preserve the original timedate stamps.  This is valuable since many rsync from the generated directory and each run these unchanged files have to be sync'd.

Thanks!

**Comment by rprieto** [comment]
It's a good idea. I wonder if there could be a strange behaviour when switching themes. Based on the date the theme was updated, the `public` folder would be updated with "earlier" files. Would this trigger an upload anyway on most systems (e.g. `s3 sync`)?

**Comment by scubachristopher** [comment]
You're correct -- there is an issue.  Rsync behaves correctly, since it detects a change in timedate and copies if a change is detected, even if it's older (not using --update, which will ignore if the dest is newer).  s3 sync does not behave this way:

> A local file will require uploading if the size of the local file is different than the size of the s3 object, the last modified time of the local file is newer than the last modified time of the s3 object, or the local file does not exist under the specified bucket and prefix.

I've tested both rsync (works) and s3 (does not work).

That's a bummer because I never change themes.  Perhaps you can let the users control this by adding a "--no_theme_update" where you just would not copy the theme content.  Or you could diff the source theme content and dest and just not update if they are the same.

**Comment by rprieto** [comment]
Hi @scubachristopher. I'm revisiting this issue and I think we can make it work with S3:

```
--size-only (boolean) Makes the size of each key the only criteria used to decide whether to sync from source to destination.

--exact-timestamps (boolean) When syncing from S3 to local, same-sized items will be ignored only when the timestamps match exactly. The default behavior is to ignore same-sized items unless the local version is newer than the S3 version.
```

My understanding is that it
- first checks the size. If different it uploads.
- if same size, it then checks the timestamp. By default it will check if it's newer. However the `--exact-timestamp` flag does what we want.

Do you mind re-running your test if you still have it handy? If it works I'm keen to make `cp -p` the default.

**Comment by gaby** [comment]
@rprieto Any way to have an auto-sync? I have multiple users uploading images, and right now that requires running the thumbsup container on each change.

Would be cool if it had a built-in thing that checks for changes and re-generates the site
